compiler/package.go

```diff
@@ -0,0 +1,207 @@
+package compiler
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/brimdata/super/compiler/dag"
+	"github.com/brimdata/super/compiler/kernel"
+	"github.com/brimdata/super/compiler/optimizer"
+	"github.com/brimdata/super/compiler/parser"
+	"github.com/brimdata/super/compiler/semantic"
+	"github.com/brimdata/super/lakeparse"
+	"github.com/brimdata/super/order"
+	"github.com/brimdata/super/runtime"
+	"github.com/brimdata/super/runtime/exec"
+	"github.com/brimdata/super/runtime/sam/op"
+	"github.com/brimdata/super/runtime/vam"
+	vamop "github.com/brimdata/super/runtime/vam/op"
+	"github.com/brimdata/super/runtime/vcache"
+	"github.com/brimdata/super/zbuf"
+	"github.com/brimdata/super/zio"
+)
+
+func Parse(query string, filenames ...string) (*parser.AST, error) {
+	return parser.ParseQuery(query, filenames...)
+}
+
+func Analyze(rctx *runtime.Context, ast *parser.AST, env *exec.Environment, extInput bool) (dag.Seq, error) {
+	return semantic.Analyze(rctx.Context, ast, env, extInput)
+}
+
+func Optimize(rctx *runtime.Context, seq dag.Seq, env *exec.Environment, parallel int) (dag.Seq, error) {
+	// Call optimize to possible push down a filter predicate into the
+	// kernel.Reader so that the zng scanner can do boyer-moore.
+	o := optimizer.New(rctx.Context, env)
+	var err error
+	seq, err = o.Optimize(seq)
+	if err != nil {
+		return nil, err
+	}
+	if parallel > 1 {
+		// For an internal reader (like a shaper on intake), we don't do
+		// any parallelization right now though this could be potentially
+		// beneficial depending on where the bottleneck is for a given shaper.
+		// See issue #2641.
+		seq, err = o.Parallelize(seq, parallel)
+		if err != nil {
+			return nil, err
+		}
+		seq, err = o.Vectorize(seq) // XXX why is this here?
```

Comment:
This is how we engage the vector runtime on lake queries when CSUP is available.